api/gateway/basecommerce/basecommerce.cfc

```diff
@@ -0,0 +1,269 @@
+					local.results = xmlParse(local.response.getResult());					
+					
+					//Store parsed result
+					local.response.setParsedResult(local.results);
+					
+					//Check returned XML for success/failure
+					if(structKeyExists(local.results.xmlRoot, 'error_response')) {
+						local.response.setStatus(getService().getStatusFailure());
+					} else {
+						local.response.setStatus(getService().getStatusSuccessful());
+					}
+				} else {
+					local.pairs = listToArray(local.response.getResult(), '&');
+					//split the variable=value
+					for(local.ii=1; local.ii <= arrayLen(pairs); local.ii++) {
+						if(listLen(local.pairs[local.ii], '=') GT 1) {
+							local.results[listFirst(local.pairs[local.ii], '=')] = listLast(local.pairs[local.ii], '=');
+						} else {
+							local.results[listFirst(local.pairs[local.ii], '=')] = '';
+						}
+                    }
+                    
+					//store parsed result
+					local.response.setParsedResult(local.results);
+	
+					//handle common response fields
+					if(structKeyExists(local.results, 'response_code')) {
+						local.response.setMessage(variables.braintree[local.results.response_code]);
+					}
+					if(structKeyExists(local.results, 'response_text')) {
+						local.response.setMessage(local.response.getMessage() & ': ' & variables.braintree[local.results.response_text]);
+					}
+					if(structKeyExists(local.results, 'transactionid')) {
+						local.response.setTransactionID(local.results.transactionid);
+					}
+					if(structKeyExists(local.results, 'authcode')) {
+						local.response.setAuthorization(local.results.authcode);
+					}
+					if(structKeyExists(local.results, 'customer_vault_id')) {
+						local.response.setTokenID(local.results.customer_vault_id);
+					}
+					
+					//handle common 'success' fields
+					if(structKeyExists(local.results, 'avsresponse')) {
+						local.response.setAVSCode(local.results.avsresponse);
+					}
+					if(structKeyExists(local.results, 'cvvresponse')) {
+						local.response.setCVVCode(local.results.cvvresponse);
+					}
+	
+					//see if the response was successful
+					if(local.results.response EQ '1') {
+						local.response.setStatus(getService().getStatusSuccessful());
+					} else if(local.results.response EQ '2') {
+						local.response.setStatus(getService().getStatusDeclined());
+					} else {
+						//only other known state is 3 meaning, 'error in transaction data or system error'
+						local.response.setStatus(getService().getStatusFailure());
+					}
+				}
+			} else {
+				//This is bad, because Braintree didn't return a response.  Uh oh!
+				local.response.setStatus(getService().getStatusUnknown());
+			}
+		}
+		
+		return local.response;
+	}
+
+	private any function createResponse() {
+		return createObject("component", "cfpayment.api.gateway.basecommerce.response").init(argumentCollection = arguments, service = getService());
+	}
+	
+	private any function CreateEmptyResponse(struct payload=structNew()) {
```

Comment:
Kill this guy and just map the values from the java response object into the cfpayment response object for status, transactionid, etc.